docs/ReferenceGuides/UnderscoredAttributes.md

```diff
@@ -0,0 +1,428 @@
+  }
+  func h(@_implicitSelfCapture _: @escaping () -> Void) {
+    h({ f() }) // ok
+  }
+}
+```
+
+## `@_inheritActorContext`
+
+(Note that it is "inherit", not "inherits", unlike below.)
+
+## `@_inheritsConvenienceInitializers`
+
+An attribute that signals that a class declaration inherits its convenience
+initializers from its superclass. This implies that all designated initializers
+-- even those that may not be visible in a swiftinterface file -- are
+overridden. This attribute is often printed alongside
+`@_hasMissingDesignatedInitializers` in this case.
+
+## `@_marker`
+
+Indicates that a protocol is a marker protocol;
+one that represents some meaningful property at compile-time
+but has no runtime representation.
+
+For more details, see [SE-0302](https://github.com/apple/swift-evolution/blob/main/proposals/0302-concurrent-value-and-concurrent-closures.md#marker-protocols), which introduces marker protocols.
+
+At the moment, the language only has 1 marker protocol: `Sendable`.
+
+Fun fact: Rust has a very similar concept called
+[marker traits](https://doc.rust-lang.org/std/marker/index.html),
+including one called `Send`,
+which inspired the design of `Sendable`.
+
+## `@_nonEphemeral`
+
+Marks a function parameter that cannot accept a temporary pointer produced from
+an inout-to-pointer, array-to-pointer, or string-to-pointer conversion. Such a
+parameter may only accept a pointer that is guaranteed to outlive the duration
+of the function call.
+
+Attempting to pass a temporary pointer to an `@_nonEphemeral` parameter will
+produce a warning. This attribute is primarily used within the standard library
+on the various `UnsafePointer` initializers to warn users about
+the undefined behavior caused by using a temporary pointer conversion as an
+argument:
+
+```swift
+func baz() {
+  var x = 0
+
+  // warning: Initialization of 'UnsafePointer<Int>' results in a dangling pointer
+  let ptr = UnsafePointer(&x)
+
+  // warning: Initialization of 'UnsafePointer<Int>' results in a dangling pointer
+  let ptr2 = UnsafePointer([1, 2, 3])
+}
+```
+
+The temporary pointer conversion produces a pointer that is only
+guaranteed to be valid for the duration of the call to the initializer,
+and becomes invalid once the call ends.
+So the newly created `UnsafePointer` will be dangling.
+
+One exception to this is that inout-to-pointer conversions
+on static stored properties and global stored properties
+produce non-ephemeral pointers, as long as they have no observers:
+
+```swift
+var global = 0
+
+struct S {
+  static var staticVar = 0
+}
+
+func baz() {
+  let ptr = UnsafePointer(&global) // okay
+  let ptr2 = UnsafePointer(&S.staticVar) // okay
+}
+```
+
+Additionally, if they are of a tuple or struct type, their stored members
+without observers may also be passed inout as non-ephemeral pointers.
+
+For more details, see [the educational note on temporary pointer usage](/userdocs/diagnostics/temporary-pointers.md).
+
+## `@_nonoverride`
```

Comment:
As per #19034:

#### `@_nonoverride`

The standard library and overlays are compiled with the `-warn-implicit-overrides` flag, which warns when a protocol restates a requirement from another protocol that it refines without annotating the declaration with either `override` or `@_nonoverride`.

An `override` annotation causes the overriding declaration to be treated identically to the overridden declaration; a conforming type can only provide one implementation ("witness"). Restating a protocol requirement and then marking it as an `override` is generally only needed to help associated type inference, and many `override` annotations correlate closely with ABI FIXMEs.

Meanwhile, `@_nonoverride` is the "opposite" of `override`, allowing two protocol requirements to be treated independently; a conforming type can provide a distinct witness for each requirement (for example, by using `@_implements`). Use `@_nonoverride` when semantics differ between the two requirements. For example, `BidirectionalCollection.index(_:offsetBy:)` allows negative offsets, while `Collection.index(_:offsetBy:)` does not, and therefore the former is marked `@_nonoverride`.

The `@_nonoverride` annotation can also be specified on class members in addition to protocol members. Since it is the "opposite" of `override`, it can be used to suppress "near-miss" diagnostics for declarations that are similar to but not meant to override another declaration, and it can be used to intentionally break the override chain, creating an overload instead of an override.